src/foam/nanos/dig/ExternalDataAware.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.INTERFACE({
+  package: 'foam.nanos.dig',
+  name: 'ExternalDataAware',
+  documentation: 'Interface for marking an object that contains the externalData map',
+
+  methods: [
```

Comment:
Shouldn't you just be able to add the single property and these methods will be generated automatically in the interface?

Reply:
@nanoscott there's a bug with using `Map` in an interface.  It includes the factory in the interface as a public method, so when the method is declared as protected in an implementation it causes the build to fail since you can't restrict the visibility of inherited methods.

Reply:
ok